internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangelogConsumerClientFactory.java

```diff
@@ -32,11 +39,32 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
       newStoreChangelogClientConfig.setD2ControllerClient(d2ControllerClient);
       newStoreChangelogClientConfig
           .setSchemaReader(ClientFactory.getSchemaReader(newStoreChangelogClientConfig.getInnerClientConfig()));
-      if (newStoreChangelogClientConfig.getViewClassName()
-          .equals(ChangeCaptureView.CHANGE_CAPTURE_VIEW_WRITER_CLASS_NAME)) {
+
+      String viewClass = getViewClass(
+          storeName,
+          newStoreChangelogClientConfig.getViewClassName(),
+          d2ControllerClient,
+          globalChangelogClientConfig.getControllerRequestRetryCount());
+      if (viewClass.equals(ChangeCaptureView.class.getCanonicalName())) {
```

Comment:
Minor: as we created the LOGGER int this class, maybe we can log here?  or we should not? (as there are some logs inside the changelog class constructor)